preload/fetch_destination.https.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../service-workers/service-worker/resources/test-helpers.sub.js"></script>
+<script>
+async_test(function(t) {
+    var worker_url = 'resources/fetch-destination-worker.js';
+    var scope = 'resources/dummy.xml';
+    var registration;
+
+    function done(registration) {
+      registration.unregister();
+      t.done();
+    }
+    service_worker_unregister_and_register(t, worker_url, scope)
+      .then(t.step_func(function(r) {
+          registration = r;
+          return wait_for_state(t, r.installing, 'activated');
+        }))
+      .then(t.step_func(function() {
+          return with_iframe(scope);
+        }))
+      .then(t.step_func(function(frame) {
+          var link = frame.contentWindow.document.createElement("link");
+          link.as = "fetch";
+          link.href = scope;
+          link.rel = "preload";
+          link.addEventListener("load", t.step_func(function() { done(registration); }));
+          link.addEventListener("error", t.step_func(function() { assert_unreached("Fetch destination preload errored"); done(registration); }));
```

Comment:
I don't think it's an issue, but I would just leave out done(). Come to think of it, you can use `t.step_func_done()` which cleans this up automatically.